accepted/0000-npm7-support-package-lock-6.md

```diff
@@ -0,0 +1,59 @@
+# Support `package-lock.json` v3 in npm 7
+
+## Summary
+
+Currently npm 7 can read `package-lock.json` versions 1, 2, and 3, but it always writes version 2. It would be nice if npm 7 could write `package-lock.json` version 3 instead.
+
+## Motivation
+
+npm 7 introduces new features which aren’t compatible with older npm versions, such as workspaces. This makes npm 7 a minimum requirement for a project.
+
+Both `package-lock.json` version 1 and 3 files can be huge. `package-lock.json` 2 is an intermediate format whose goal is to add backwards compatibility with `package-lock.json` version 1 and npm 6. It does so by writing both the dependency trees in the version 1 and 3 formats. This means the file is even twice as big as it needs to be. This also means the same change appears twice in a diff, but in a different format, meaning it needs to be reviewed twice. This is unnecessary if the minimal required npm version is 7.
+
+## Detailed Explanation
+
+npm is expected to output the version of `package-lock.json` which is configured in `.npmrc`.
+
+If `.npmrc` contains the following line, `package-lock.json` version 1 is used.
+
+```ini
+package-lock-version=1
+```
+
+If `.npmrc` contains the following line, `package-lock.json` version 2 is used.
+
+```ini
+package-lock-version=2
+```
+
+If `.npmrc` contains the following line, `package-lock.json` version 3 is used.
+
+```ini
+package-lock-version=3
+```
+
+Major npm versions may choose to drop support for old formats.
+
+## Rationale and Alternatives
+
+1. A command line flag could be supported instead of an option in `.npmrc`. However, typically one would want to support this on a project level.
```

Comment:
My understanding is that typically command line flags are _also_ usable in `.npmrc` - so supporting `npm install --package-lock-version=<number>` would also allow you to use `package-lock-version=<number>` in `.npmrc`.

IMO the path of both is ideal.